utils/txHandler.ts

```diff
@@ -139,77 +137,80 @@ export const signAndWaitTx = async (
-                    const seed_bytes = Uint8Array.from(Buffer.from(json_response["seed"].substring(2), 'hex'));
-                    let shuffled_extrinsics = recreateExtrinsicsOrder(prev_block_extrinsics, seed_bytes);
-
-                    // filter extrinsic triggered by current request
-                    let index = shuffled_extrinsics.findIndex( e => {return e.isSigned && e.signer.toString() === who.address && e.nonce.toString() === nonce.toString();});
-                    if (index < 0) {
-                        return;
+      const api = getApi()
+      let result: GenericEvent[] = [];
+    
+      if (timeout_ms > 0){
+          setTimeout(() => {
+          reject(`W[${env.JEST_WORKER_ID}] timeout in - signAndWaitTx - " + ${who.address} + " - " + ${nonce} + " - " + ${JSON.stringify(tx.toHuman())}` )
+          }, timeout_ms);
+      }
+    
+      const unsub = await tx.signAndSend(who, { nonce }, async ({ events = [], status }) => {
+        //console.warn(status);
+        if (status.isInBlock) {
+                const unsub_new_heads = await api.derive.chain.subscribeNewHeads(async (lastHeader) => {
+                    if (lastHeader.parentHash.toString() === status.asInBlock.toString()){
+                        unsub_new_heads()
+                        let prev_block_extrinsics = (await api.rpc.chain.getBlock(lastHeader.parentHash)).block.extrinsics;
+                        let curr_block_extrinsics = (await api.rpc.chain.getBlock(lastHeader.hash)).block.extrinsics;
+                        let curr_block_events = await api.query.system.events.at(lastHeader.hash);
+    
+                        let extrinsic_with_seed = curr_block_extrinsics.find( e => { return e.method.method === "set" && e.method.section === "random" });
+                        if(!extrinsic_with_seed){
+                            return;
+                        }
+    
+                        var json_response = JSON.parse(extrinsic_with_seed.method.args[0].toString())
+                        const seed_bytes = Uint8Array.from(Buffer.from(json_response["seed"].substring(2), 'hex'));
+                        let shuffled_extrinsics = recreateExtrinsicsOrder(prev_block_extrinsics, seed_bytes);
+    
+                        // filter extrinsic triggered by current request
+                        let index = shuffled_extrinsics.findIndex( e => {return e.isSigned && e.signer.toString() === who.address && e.nonce.toString() === nonce.toString();});
+                        if (index < 0) {
+                            return;
+                        }
+                        
+    
+                        let req_events = curr_block_events.filter(event => {
+                            return event.phase.isApplyExtrinsic && event.phase.asApplyExtrinsic.toNumber() === index;
+                        }).map( ({ phase, event }) => {
+                            return event;
+                        });
+                        console.log(
```

Comment:
shouldnt that be enable on some env variable or switch

Reply:
Right! I'll review all the traces and handle extra debug options accordingly.

Reply:
Added a logging framework,
```
[2021-08-02T11:50:39.959Z] - W[3] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:39.949Z] - W[4] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.020Z] - W[1] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.001Z] - W[5] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.029Z] - W[6] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:39.990Z] - W[7] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:40.028Z] - W[2] - [info]: TEST_INFO: Running test in ws://127.0.0.1:9944
[2021-08-02T11:50:41.935Z] - W[1] - [debug]: 0x41b2b6ce07a6aab7dd420715ab63ec1b8b06f17d7f8d67b5599e15cce53b4631
[2021-08-02T11:50:41.954Z] - W[1] - [info]: Chain is at block: #470
[2021-08-02T11:50:42.021Z] - W[1] - [info]: Chain is at block: #470
```
/mangata-e2e/utils/Logger.ts
* Which define custom levels based on the env.Variable.
* Write into a file ( that will be exported in the actions )
* Defined a custom formatting! 

```
Test Suites: 18 passed, 18 total
Tests:       86 passed, 86 total
Snapshots:   0 total
Time:        928.571 s
```
Thanks for the hint @mateuszaaa !